typing/jkind.mli

```diff
@@ -524,8 +524,17 @@ val get_annotation : 'd Types.jkind -> Parsetree.jkind_annotation option
 (*********************************)
 (* normalization *)
 
+type 'd normalize_mode =
```

Comment:
The indices on the two constructors look backwards. I would expect `Require_best` to be happy with either a left or a right, but `Ignore_best` should only work on the left. I'm worried I'm missing something if everything has type-checked with things as written.

Reply:
Ah, well, this type is used only on the left, anyway (in `normalize`, which takes a `jkind_l`). But then I think that means it doesn't need to be indexed.

Reply:
I think you have the meaning of the indexing backwards. It's specifying the result of normalization, not restricting the input type. There are a few places that require there to be no with-bounds after `Ignore_best` normalization, at this gives us a type assurance that this is the case.

Reply:
In fact, making this type indexed helped us discover a bug in normalization.

Reply:
Ah, I see what you mean. Yes, you're right that this shouldn't leak into the mli. I'll fix.